### PR TITLE
[PLATFORM-442] Finalize stream endpoints

### PR DESCRIPTION
**Pull request by fonty1**
Added support for storageDays, requireSignedData and autoConfigure on streams.

Works for Stream POSTs (creating), waiting on endpoint: https://github.com/streamr-dev/engine-and-editor/pull/588

For it to work with PUTs (editing)

### CODE REVIEW

File: app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx [hunk 1]
```diff
@@ -241,25 +244,31 @@ export class ConfigureView extends Component<Props, State> {
                 }
                 <div className={styles.settings}>
                     {/* eslint-disable jsx-a11y/label-has-associated-control */}
+                    {stream && stream.autoConfigure !== undefined &&
+                        <Fragment>
```
Comment (tumppi) [inline app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx]: Fragment should not be needed here?

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx [hunk 2]
```diff
@@ -316,6 +319,7 @@ class HistoryView extends Component<Props, State> {
                         </Row>
                     </Fragment>
                 )}
+                {stream && stream.storageDays &&
```
Comment (tumppi) [inline app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: Maybe check for `!== undefined` here so it works if `storageDays` is zero?

Comment (fonty1) [reply app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: ahh, really good point! Will fix.

File: app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx (hunk 1 above)
Comment (fonty1) [reply app/src/userpages/components/StreamPage/Show/ConfigureView/index.jsx]: ah yep, looks like some overcooked copy pasta. Cheers

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx [hunk 3]
```diff
@@ -329,12 +331,12 @@ class HistoryView extends Component<Props, State> {
                             id="storage-period"
                             type="number"
                             label={I18n.t('userpages.streams.edit.configure.historicalStoragePeriod.label')}
-                            value={historicalStoragePeriod}
+                            value={stream.storageDays ? stream.storageDays : 365}
```
Comment (timoxley) [inline app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: can storageDays be `0`?

File: app/src/userpages/modules/userPageStreams/actions.js [hunk 4]
```diff
@@ -502,6 +502,11 @@ export const initEditStream = () => (dispatch: Function, getState: Function) =>
             config: cloneDeep(stream.config) || {},
             ownPermissions: cloneDeep(stream.ownPermissions) || [],
             lastUpdated: stream.lastUpdated || 0,
+            autoConfigure: stream.autoConfigure || false,
+            partitions: stream.partitions || 1,
+            requireSignedData: stream.requireSignedData || false,
+            storageDays: stream.storageDays || 365,
```
Comment (timoxley) [inline app/src/userpages/modules/userPageStreams/actions.js]: zero?

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx (hunk 3 above)
Comment (hpihkala) [reply app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: yes

File: app/src/userpages/modules/userPageStreams/actions.js (hunk 4 above)
Comment (hpihkala) [reply app/src/userpages/modules/userPageStreams/actions.js]: can be zero

File: app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx (hunk 3 above)
Comment (fonty1) [reply app/src/userpages/components/StreamPage/Show/HistoryView/index.jsx]: thanks guys

**Pull request merged**